### PR TITLE
Added encrypted passwords and user authentication using bcrypt

### PR DESCRIPTION
**Pull request by clovellbsc**
## Before you start
1. Is there an issue on the repository for this request? If not, please create one.
2. Have you followed the correct workflow in the [team documentation](https://docs.google.com/document/d/13yi709ax4nmfTG9J0F57yEFpYRH7fdtczwvVJCrNIsM/edit#)?
3. Conflict resolution to be dealt with **on GitHub**.

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have created a PR ticket on the [Trello board](https://trello.com/b/P8lEsvZe/acebook).
- [x] I will link the PR to the Trello card once created.
- [x] I will send the PR in the Slack channel and ping @here.

### CODE REVIEW

**Comment by clovellbsc** [comment]
[Encryption of password and authentication](https://trello.com/c/j23oaABl/55-encryption-of-password-and-authentication)

File: controllers/sessions.js [hunk 1]
```diff
@@ -9,15 +10,21 @@ const SessionsController = {
     console.log("trying to log in");
     const email = req.body.email;
     const password = req.body.password;
-
+    console.log(password);
```
Comment (PKilgarriff) [inline controllers/sessions.js:13]: Don't think this should be left in, even temporarily

File: controllers/sessions.js [hunk 2]
```diff
@@ -9,15 +10,21 @@ const SessionsController = {
...
     User.findOne({ email: email }).then((user) => {
-      if (!user) {
-        res.redirect("/sessions/new");
-      } else if (user.password != password) {
-        res.redirect("/sessions/new");
+      if (user) {
+        bcrypt.compare(password, user.password).then((result) => {
+          if (result) {
+            console.log(`Look here -------- ${user.password}`);
```
Comment (PKilgarriff) [inline controllers/sessions.js:18]: Same as above -> worth removing just on the off-chance we forget about it

**Review by PKilgarriff — CHANGES_REQUESTED** [review]
All looks good - just recommend getting rid of those password console.logs before we merge

File: controllers/sessions.js (hunk 2 above)
Comment (adamwoodcock98) [reply controllers/sessions.js]: These have been removed by Chris.

**Review by PKilgarriff — APPROVED** [review]
Cheers, like the error in handlebars as well

**Review by adamwoodcock98 — APPROVED** [review]
Excellent work!

**Pull request merged**